tests/suites/test_suite_pk.data

```diff
@@ -9,7 +9,7 @@ depends_on:MBEDTLS_RSA_C
 valid_parameters_pkwrite:"308204a20201000282010100a9021f3d406ad555538bfd36ee82652e15615e89bfb8e84590dbee881652d3f143504796125964876bfd2be046f973beddcf92e1915bed66a06f8929794580d0836ad54143775f397c09044782b0573970eda3ec15191ea8330847c10542a9fd4cc3b4dfdd061f4d1051406773130f40f86d81255f0ab153c6307e1539acf95aee7f929ea6055be7139785b52392d9d42406d50925897507dda61a8f3f0919bead652c64eb959bdcfe415e17a6da6c5b69cc02ba142c16249c4adccdd0f7526773f12da023fd7ef431ca2d70ca890b04db2ea64f706e9ecebd5889e253599e6e5a9265e2883f0c9419a3dde5e89d9513ed29dbab7012dc5aca6b17ab528254b10203010001028201001689f5e89142ae18a6ffb0513715a4b0b4a13b9e5b3729a2bd62d738c6e15cea7bf3a4d85ab2193a0628c9452bb1f0c1af8b132789df1c95e72778bf5330f5b0d915d242d5e0818e85001ed5fa93d1ce13455deb0a15438562e8e3c8d60ec1e4c9ebff9f2b36b9cde9332cc79f0d17a7ae79cc1353cd75409ad9b4b6d7ee3d82af6f3207656cf2ac98947c15c398db0cebf8dc3eef5398269480cdd09411b960273ae3f364da09af849f24aa87346c58618ea91d9d6cd1d3932c80dbfc1f0a4166a9036911999ca27761079f0ce02db02c1c909ff9b4278578d7bb1b54b2b7082fc9e864b6b394e331c0d11a9a68255565b6dd477f4119c5809839520700711102818100d7db987ad86de6a9b0749fb5da80bacde3bebd72dcc83f60a27db74f927ac3661386577bfce5b4a00ad024682401d6aad29713c8e223b53415305ca07559821099b187fdd1bad3dc4dec9da96f5fa6128331e8f7d89f1e1a788698d1a27256dc7cd392f04e531a9e38e7265bf4fd7eec01e7835e9b1a0dd8923e440381be1c2702818100c87025fff7a493c623404966fbc8b32ed164ca620ad1a0ad11ef42fd12118456017856a8b42e5d4ad36104e9dc9f8a2f3003c3957ffddb20e2f4e3fc3cf2cdddae01f57a56de4fd24b91ab6d3e5cc0e8af0473659594a6bbfdaacf958f19c8d508eac12d8977616af6877106288093d37904a139220c1bc278ea56edc086976702818043e708685c7cf5fa9b4f948e1856366d5e1f3a694f9a8e954f884c89f3823ac5798ee12657bfcaba2dac9c47464c6dc2fecc17a531be19da706fee336bb6e47b645dbc71d3eff9856bddeb1ac9b644ffbdd58d7ba9e1240f1faaf797ba8a4d58becbaf85789e1bd979fcfccc209d3db7f0416bc9eef09b3a6d86b8ce8199d4310281804f4b86ccffe49d0d8ace98fb63ea9f708b284ba483d130b6a75cb76cb4e4372d6b41774f20912319420ca4cbfc1b25a8cb5f01d6381f6ebc50ed3ef08010327f5ba2acc1ac7220b3fa6f7399314db2879b0db0b5647abd87abb01295815a5b086491b2c0d81c616ed67ef8a8ce0727f446711d7323d4147b5828a52143c43b4b028180540756beba83c20a0bda11d6dec706a71744ff28090cec079dffb507d82828038fe657f61496a20317f779cb683ce8196c29a6fe28839a282eef4de57773be56808b0c3e2ac7747e2b200b2fbf20b55258cd24622a1ce0099de098ab0855106ae087f08b0c8c346d81619400c1b4838e33ed9ff90f05db8fccf8fb7ab881ca12"
 
 PK utils: RSA Minimum key
-depends_on:MBEDTLS_RSA_C:MBEDTLS_GENPRIME
+depends_on:MBEDTLS_RSA_C:MBEDTLS_GENPRIME:!MBEDTLS_VALGRIND
```

Comment:
As far as I can see, the sole rationale for skipping some test cases is that they're slow. I am not seeing any considerations on the tests' usefulness.

For each family of tests that is becoming skipped, please indicate why it's ok to skip them, i.e. why running those tests with valgrind is unlikely to catch bugs that would be missed by only running the remaining tests with valgrind and the now-skipped tests without valgrind.

Reply:
The rationale is indeed, skip the slow tests (but try to retain equivalent tests over smaller inputs where possible). There is a cost/benefit for every test that we run, and the cost becomes very high in some cases, whereas the benefit is very small in almost all cases, given that we have ASan and MSan.

Outside of the constant-flow tests, I'm not convinced Valgrind adds very much value (can you remember the last bug caught by Valgrind that wasn't caught by Asan/Msan?). I do think that faster CI adds considerable value.

Reply:
> can you remember the last bug caught by Valgrind that wasn't caught by Asan/Msan?

I remember that it happened once, a long time ago. And that we didn't investigate fully, and it might have been due to some different compiler optimization that made the Msan/Asan build not have the bug at runtime, or due to how Msan/Asan sometimes give weird results when inline assembly is resolved.

I think that's an investigation worth doing. I disagree with the reasoning that we just skip slow tests with Valgrind without regard for coverage. I would agree with removing all Valgrind tests if we determine that as a whole, they have a negligible, perhaps null benefit to coverage.

Reply:
>I would agree with removing all Valgrind tests if we determine that as a whole, they have a negligible, perhaps null benefit to coverage.

I'm not sure I'm comfortable with removing all valgrind tests at this point, but I certainly agree with the idea of treating all valgrind testing uniformly, be it "keep", "move to nightly-only" or "remove altogether".

(Note we can also take an incremental approach where we first move to nightly-only, then remove altogether after some time has passed and nothing was found by the nightly that had been missed by the PR jobs.)